src/server/api/middleware/auth.js

```diff
@@ -7,6 +9,33 @@ function authenticateV2Login(req, res, next) {
   next();
 }
 
+function checkParamWalletMatch(req, res, next) {
+  const { user } = req.session;
+  const { wallet } = req.params;
+  if (user !== wallet) {
+    res.status(400).send('ADDRESS_MISMATCH');
+    return;
+  }
+  next();
+}
+
+async function checkEmailHasVerified(req, res, next) {
+  const { wallet } = req.params;
+  const userDoc = await walletUserCollection.doc(wallet).get();
+  const { email, emailUnconfirmed } = userDoc.data();
```

Comment:
Do we need to guard `userDoc.data()` not exists?

Reply:
Logically we should have created a doc for a logged-in user, else the request would be blocked by the `authenticateV2Login` middleware before calling `checkEmailHasVerified`?

Reply:
But that's the case if we always check `authenticateV2Login` first... 🌚